Manifest dt

jenswi-linaro opened the pull request:
<!--
    If you are new to submitting pull requests to OP-TEE, then please have a
    look at the list below and tick them off before submitting the pull request.

    1. Read our contribution guidelines:
         https://optee.readthedocs.io/en/latest/general/contribute.html

    2. Read the contribution section in Notice.md and pay extra attention to the
       "Developer Certificate of Origin" in the contribution guidelines.

    3. You should run checkpatch preferably before submitting the pull request.

    4. When everything has been reviewed, you will need to squash, rebase and
       add tags like `Reviewed-by`, `Acked-by`, `Tested-by` etc. More details
       about this can also be found on the link provided above.

    NOTE: This comment will not be shown in the pull request, so no harm keeping
    it, but feel free to remove it if you like.
-->

etienne-lms commented on core/arch/arm/kernel/boot.c:
```diff
@@ -1165,12 +1171,18 @@ static bool cpu_nmfi_enabled(void)
  */
 void __weak boot_init_primary_late(unsigned long fdt,
-				   unsigned long tos_fw_config)
+				   unsigned long manifest_fdt)
```
This local `manifest_fdt` and the global `manifest_dt` quite look alike, it a bit confusing.

jenswi-linaro replied:
I'll think of something.

etienne-lms reviewed (commented):
The second commit does many things. I felt a bit lost in the implication of requiring a manifest DTB in all the SPMC configurations. Is it always read-only, even when `CFG_CORE_SEL1_SPMC=y`?

jenswi-linaro commented on core/mm/core_mmu.c:
```diff
@@ -755,6 +756,8 @@ uint32_t core_mmu_type_to_attr(enum teecore_memtypes t)
 	case MEM_AREA_NEX_NSEC_SHM:
 		return attr | TEE_MATTR_PRW | cached;
+	case MEM_AREA_MANIFEST_DT:
+		return attr | TEE_MATTR_SECURE | TEE_MATTR_PR;
```
Something is missing, it should be `cached` since we have coherent memory when using this. I'll fix.

jenswi-linaro commented:
> The second commit does many things. I felt a bit lost in the implication of requiring a manifest DTB in all the SPMC configurations. Is it always read-only, even when `CFG_CORE_SEL1_SPMC=y`?

I'm sorry, I'll try to break it into smaller pieces. Yes, the manifest is always supposed to be read-only, it's not to be confused with the system DTB.

jenswi-linaro commented:
I've split "core: rename tos_fw_config_dt to manifest_dt" into multiple commits.
I've also added "libutils: add runtime_assert()" for better stubbing of `core_mmu_set_secure_memory()` depending on configuration.

LeiSen62 commented:
Acked-by: Leisen <leisen1@huawei.com>

etienne-lms reviewed (commented):
`Acked-by: Etienne Carriere <etienne.carriere@foss.st.com>` for the series.

jenswi-linaro commented:
Tags applied.

balint-dobszay-arm commented:
Hi Jens,
I get a crash when `CFG_CORE_SEL1_SPMC=y` and `CFG_CORE_TPM_EVENT_LOG=y`. After your changes the manifest DT is mapped as RO: https://github.com/OP-TEE/optee_os/pull/6289/commits/aca027763ffaf76c8a36ae787a085b402d212185#diff-e50fbd95497200fea2a8eb3e91c406c07bc915bd20bdc6d76b88bd7c06ee1a74R760

But the TPM event log handling tries to write to it here: https://github.com/OP-TEE/optee_os/blob/master/core/kernel/tpm.c#L50

Making the region RW and/or removing the mentioned `fdt_setprop()` solves the issue. When the DT is in S memory I don't really see a reason to explicitly zero the event log address, so probably removing the write is the better solution.
What do you think?

jenswi-linaro commented:
I agree, removing the writing makes sense. Do you mind making a PR to fix it or shall I do it?

balint-dobszay-arm commented:
I'll create a PR. It looks like this only affects the S-EL1 SPMC use case, since otherwise `tpm_map_log_area()` operates on the external DT which is RW.

jenswi-linaro commented:
Thank you

balint-dobszay-arm commented:
FYI https://github.com/OP-TEE/optee_os/pull/6310
